samples/bluetooth/peripheral_hids_keyboard/README.rst

```diff
@@ -0,0 +1,145 @@
+   To bond, click the settings button for the device in nRF Connect, select **Pair**, check **Perform Bonding**, and click **Pair**.
+   Wait until the bond is established before you continue.
+   Observe that the services are shown in the connected device.
+#. Push Button 1 on the board.
+   Observe that two notifications are received on one of the HID Report characteristics, denoting press and release for one character of the test message.
+
+   The first notification has the value ``00000B0000000000``, the second has the value ``0000000000000000``.
+   The format used for keyboard reports is the following byte array: ``[modifier, reserved, Key1, Key2, Key3, Key4, Key6, Key7]``.
+   The received values correspond to press and release of character "h".
+
+   Similarly, further press and release events will result in press and release notifications of subsequent characters of the test string.
+   Therefore, pushing Button 1 again will result in notification of press and release reports for character "e".
+#. Push Button 2 and hold it while pushing Button 1.
+   Pushing Button 2 changes the modifier bit to 02, which simulates pushing the Shift key on a keyboard.
+
+   Observe that two notifications are received on one of the HID Report characteristics, denoting press and release for one character of the test message.
+   The first one has the value ``02000F0000000000``, the second has the value ``0200000000000000``.
+   These values correspond to press and release of character "l" with the Shift key pressed.
+#. In nRF Connect, select the HID Report (which has UUID 0x2A4D and the properties Read, WriteWithoutResponse, and Write).
+   Enter ``02`` in the text box and click the **Write** button.
+   This sets the modifier bit of the Output Report to 02, which simulates turning Caps Lock ON.
+
+   Observe that LED 1 turns on.
+#. Select the same HID Report again.
+   Enter ``00`` in the text box and click the **Write** button.
+   This sets the modifier bit to 00, which simulates turning Caps Lock OFF.
+
+   Observe that LED 1 turns off.
+#. Disconnect the device in nRF Connect.
+   Observe that no new notifications are received and the device is advertising.
+#. As bond information is preserved by nRF Connect, you can immediately reconnect to the device by clicking the Connect button again.
+
+
+Dependencies
+************
+
+This sample uses the following |NCS| libraries:
+
+* :ref:`hids_readme`
+
+In addition, it uses the following Zephyr libraries:
+
+* ``include/zephyr/types.h``
+* ``include/misc/printk.h``
+* ``include/misc/byteorder.h``
+* :ref:`GPIO Interface <zephyr:io_interfaces>`
+* :ref:`zephyr:settings`
+* :ref:`zephyr:bluetooth_api`:
+
+  * ``include/bluetooth/bluetooth.h``
+  * ``include/bluetooth/hci.h``
+  * ``include/bluetooth/conn.h``
+  * ``include/bluetooth/uuid.h``
+  * ``include/bluetooth/gatt.h``
+  * ``samples/bluetooth/gatt/bas.h``
+  * ``include/bluetooth/services/hids.h``
+
```

Comment:
Are these all files worth mentioning or did I miss any?
What about soc.h - I couldn't figure out which file that is.

Reply:
zephyr/soc/arm/nordic_nrf/nrf52/soc.h. But I do not think it is worth to mention it as it should be included anyway.